lib/index.js

```diff
@@ -145,6 +145,17 @@ module.exports = {
     let isEmberSource = this.project.name() === 'ember-source';
     let babelHelperPlugin = injectBabelHelpers(isEmberSource);
 
+    let compilerPath;
+    if (isEmberSource) {
+      // Here we are using the template compiler by directly evaluating it in
+      // node, because we *are* the build that produces
+      // ember-template-compiler.js.
+      compilerPath = path.resolve(__dirname, '../broccoli/glimmer-template-compiler');
+    } else {
+      // When we're building an app, we use the already built template compiler.
+      compilerPath = path.resolve(__dirname, '../dist/ember-template-compiler.js');
```

Comment:
This keeps the exact old behavior when building apps, and only changes the behavior when ember is building itself.